Automattic/wp-calypso
# Drop fallback target

#### Background

Calypso has the concept of `target` when building the app. We use `target=evergreen` to target browsers that have [good support for modern JavaScript features](https://github.com/Automattic/wp-calypso/blob/trunk/package.json#L35), and `target=fallback` to target [older browsers](https://github.com/WordPress/gutenberg/blob/33e8edc3d708e7027e787c608b1bdae65fc5d585/packages/browserslist-config/index.js) like IE 11. The main reason is that bundlers built for `evergreen` are usually smaller and faster, allowing us to deliver a better experience for the majority of our users.

Deprecating IE 11 (p7H4VZ-2T2-p2) would mean that both targets are basically the same, therefore it doesn't make sense to keep two different targets around. Dropping multi-target support will simplify many parts of our codebase.

#### Changes proposed in this Pull Request

* Drop double target for Calypso app.
* Drop any mention to `target` in the codebase:
  * The app will be served from `/calypso/app/` instead of `/calypso/<target>`
  * Images will be served from `/calypso/images/` instead of `/calypso/<target>/images`
  * Translations will be served from `/calypso/languages/` instead of `/calypso/<target>/languages`
* Drop check that redirects old browsers to `fallback`
* Rename target `evergreen` to `calypso`. We will have two targets: `wpcom` for WPCOM plugins and `calypso` for Calypso app. The goal is to allow Calypso to drop IE11 support without waiting for WPCOM.
* Expand `evergreen` (now `calypso`) to include `Chrome 80`. Chrome 80 is used for in our E2E infrastructure (specifically, WP Desktop uses  Chrome 80, Calypso E2E tests use Chrome 84). This will increase the bundle size as it will serve "less modern" JavaScript. But without this, E2E tests will break.

#### Testing instructions

* Verify all tests pass.
* Load the live branch and smoke tests everything. Be sure to switch languages as well.

#### Open questions

- What do we want to do if an unsupported browser loads Calypso? Maybe redirect them to https://browsehappy.com/? If so, we have to build that functionality before merging (or as part of) this PR.
